linera-service/tests/wasm_end_to_end_tests.rs

```diff
@@ -473,7 +475,17 @@ async fn test_wasm_end_to_end_same_wallet_fungible(config: impl LineraNetConfig)
     let (mut net, client1) = config.instantiate().await.unwrap();
 
     let chain1 = client1.get_wallet().unwrap().default_chain().unwrap();
-    let chain2 = ChainId::root(2);
+    // Get a chain different than the default
+    let mut chain2 = None;
+    for chain_id in client1.get_wallet().unwrap().chain_ids() {
+        if chain_id != chain1 {
+            chain2 = Some(chain_id);
+            break;
+        }
+    }
+
+    assert!(chain2.is_some());
+    let chain2 = chain2.expect("chain2 should be set at this point");
```

Comment:
Idea: I think this could be collapsed into

```suggestion
    let chain2 = client1
        .get_wallet()
        .unwrap()
        .chain_ids()
        .into_iter()
        .find(|chain_id| chain_id != &chain1)
        .expect("Failed to obtain a chain ID from the wallet");
```

Reply:
[![mind-blown](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/HlciHFAoHZW62zn13apJ/72765262-6caf-48ac-abe9-93ff2322d28d.gif)](https://app.graphite.dev/macros/linera-io)